### PR TITLE
libc: minimal: math: sqrt: fix Incorrect minimal libc sqrt

### LINKED ISSUE #55962: Incorrect minimal libc sqrt
**Describe the bug**
Minimal Libc SQRT has extremely low number of iterations (6), which leads to incorrect results.

https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/libc/minimal/source/math/sqrt.c#L18

`sqrtf` works fine because it targets precision, but float is sometimes not enough.

**To Reproduce**
Take `sqrt` of any sufficiently small value.
In my case, it was 0.0000016483
Obviously, `sqrt` from libc minimal should be used (`CONFIG_MINIMAL_LIBC`)

**Expected behavior**
Precision of `sqrt` results are higher than `sqrtf`.

**Impact**
Had to use my own function (locally and/or maintain own fork).

**Additional context**
I'm by no means an expert on sqrt implementations (even though I have basic understanding of Taylor series, and I can say that 6 entries is not serious), but as a quick fix I'd suggest just taking the algorithm from sqrtf and replacing float with double.

### PR DESCRIPTION
**Pull request by lawrencek52**
I crashed my computer and I am having difficulty with git getting back to where I was, but I think this code is correct.

---

Continuation of https://github.com/zephyrproject-rtos/zephyr/pull/56881

Fixes #55962

### CODE REVIEW

**Review by keith-packard — CHANGES_REQUESTED** [review]
The code is looking good, but there are a bunch of formatting issues. Get this past checkpatch and the CI and it should be ready to merge.

I'm thinking we might want to borrow tests from picolibc to validate this code during CI too; I'll see if I can find a bit of time to pull that out separately.

File: tests/lib/c_lib/src/test_sqrt.c [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+			root_squared = resf * resf;
+			zassert_true((resf > 0.0f) && (resf < INFF),
+					"sqrtf out of range");
+			if ((resf > 0.0f) && (resf < INFF)) {
+				error = (square - root_squared) /
+					square * 100;
+				if (error < 0.0) {
+					error = -error;
+				}
+				/* square and root_squared should be almost identical
```
Comment (keith-packard) [inline tests/lib/c_lib/src/test_sqrt.c:145]: This comment seems incorrect here -- you're using abs(a-b) instead of xor.

**Review by keith-packard — APPROVED** [review]
Looks good to me. I noted a couple of slightly incorrect comments in the test.

**Comment by XenuIsWatching** [comment]
@lawrencek52 Please fix the compliance issues listed in the CI failures

**Comment by keith-packard** [comment]
> @lawrencek52 Please fix the compliance and coding guidline issues listed in the CI failures

The coding guideline error refers to defining the reserved identifier 'sqrt', which this patch kinda needs to do. The compliance checks look to be commit message formatting errors which should be fixed.

**Comment by lawrencek52** [comment]
Hi Keith:

I did fix the commit formatting errors but for some reason Git hasn't seen
these changes. My level of skill with Git is VERY low and I am not sure
what I did wrong, or how to go back and fix the errors.

-Lawrence-

**Comment by keith-packard** [comment]
> Hi Keith: I did fix the commit formatting errors but for some reason Git hasn't seen these changes. My level of skill with Git is VERY low and I am not sure what I did wrong, or how to go back and fix the errors.

Yeah, it's a complicated operation. I've generated a single commit which contains all of the changes in this series along with a reformatted commit message and stuck that in my Zephyr tree here: https://github.com/keith-packard/zephyr/tree/55962_fix_typo -- I did that by using `git rebase -i e824b24017ed9b037fca844379111ceaf70f5097` and then editing the provided file to change the 'pick' to 'squash' for all but the first commit. That presented me with another file where I edited the commit message to make it conform to Zephyr requirements (I think). If you're happy with the result, I can put this into your repository with `git push -f lawrencek52 55962_fix_typo`. That should get it ready for merging.

**Comment by keith-packard** [comment]
Btw, I just tried enabling the sqrt/sqrtf tests on all targets and it works correctly for everything *except* picolibc under x86 with soft floats and leon3. Once I've got picolibc working for that configuration, I'll submit a PR that enables the tests everywhere -- Zephyr should have floating point support on all targets, although without an FPU you will end up using the soft float support provided by the toolchain.

**Comment by keith-packard** [comment]
> The changes look good, please go ahead with pushing it into my repository. What do I need to do to finish it up?

We should be all set. As you learn more about git, I suspect you'll figure out how to do the rebase and push. It's a slightly advanced git feature that lets you do development in tiny increments (as you had done) and then clean things up before they get pulled into the project. I'd bet there are plenty of tutorials on how this all works.

**Review by cfriedt — APPROVED** [review]
Thanks for the fix - the compliance failure is a false positive.

**Pull request merged**